src/boot/bootctl-install.c

```diff
@@ -449,7 +449,84 @@ static int copy_one_file(const char *esp_path, const char *name, bool force) {
         return ret;
 }
 
-static int install_binaries(const char *esp_path, const char *arch, bool force) {
+static int install_binaries_extra(const char *esp_path, const char *arch, bool force, char** ret_default_loader_path) {
+        char *root = IN_SET(arg_install_source, ARG_INSTALL_SOURCE_AUTO, ARG_INSTALL_SOURCE_IMAGE) ? arg_root : NULL;
+        _cleanup_closedir_ DIR *d = NULL;
+        _cleanup_free_ char *path = NULL;
+        _cleanup_free_ char *default_loader_path = NULL;
+        int r;
+
+        r = chase_and_opendir(BOOTLIBDIR "/extra", root, CHASE_PREFIX_ROOT|CHASE_PROHIBIT_SYMLINKS, &path, &d);
+        /* If we had a root directory to try, we didn't find it and we are in auto mode, retry on the host */
+        if (r == -ENOENT && root && arg_install_source == ARG_INSTALL_SOURCE_AUTO)
+                r = chase_and_opendir(BOOTLIBDIR "/extra", NULL, CHASE_PREFIX_ROOT|CHASE_PROHIBIT_SYMLINKS, &path, &d);
+        if (r < 0)
+                return r; /* ignore if vendor dir does not exist */
```

Comment:
should turn -ENOENT in 0 here, since a missing extra dir should not be considered an error.

and for the other errors should really log here.